File: python/task3a/S3_OBJ_01.py
def get(obj, path, fallback=None):
    current = obj
    for key in path.split("."):
        if isinstance(current, dict) and key in current:
            current = current[key]
        else:
            return fallback
    return current

data = {"a": {"b": {"c": 42}}}
print(get(data, "a.b.c", "X"))
print(get(data, "a.b.x", "fallback"))

File: js/task3a/S3_OBJ_01.js
function get(obj, path, fallback) {
  const keys = path.split(".");
  let current = obj;

  for (const key of keys) {
    if (current == null || !(key in current)) {
      return fallback;
    }
    current = current[key];
  }
  return current;
}

const data = { a: { b: { c: 42 } } };
console.log(get(data, "a.b.c", null));
console.log(get(data, "a.x.c", "missing"));
